src/txn/txn_rollback_to_stable.c

```diff
@@ -152,43 +152,45 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
     WT_DECL_ITEM(key);
     WT_DECL_RET;
     WT_ITEM full_value;
-    WT_UPDATE *hs_upd, *upd;
-    wt_timestamp_t durable_ts, hs_start_ts, hs_stop_ts;
-#ifdef HAVE_DIAGNOSTIC
-    wt_timestamp_t newer_hs_ts;
-#endif
+    WT_UPDATE *hs_upd, *tombstone, *upd;
+    wt_timestamp_t hs_durable_ts, hs_start_ts, hs_stop_durable_ts, newer_hs_durable_ts;
     uint64_t hs_counter, type_full;
     uint32_t hs_btree_id, session_flags;
     uint8_t type;
     int cmp;
     char ts_string[4][WT_TS_INT_STRING_SIZE];
     bool is_owner, valid_update_found;
+#ifdef HAVE_DIAGNOSTIC
+    bool first_record;
+#endif
 
     hs_cursor = NULL;
     hs_upd = upd = NULL;
-    durable_ts = hs_start_ts = WT_TS_NONE;
-#ifdef HAVE_DIAGNOSTIC
-    newer_hs_ts = WT_TS_NONE;
-#endif
+    hs_durable_ts = hs_start_ts = hs_stop_durable_ts = WT_TS_NONE;
     hs_btree_id = S2BT(session)->id;
     session_flags = 0;
     is_owner = valid_update_found = false;
+#ifdef HAVE_DIAGNOSTIC
+    first_record = true;
+#endif
 
     /* Allocate buffers for the data store and history store key. */
     WT_RET(__wt_scr_alloc(session, 0, &key));
     WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
     WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
 
     WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+    unpack = &_unpack;
+    WT_TIME_WINDOW_INIT(&unpack->tw);
 
     /* Get the full update value from the data store. */
     WT_CLEAR(full_value);
     if (!__wt_row_leaf_value(page, rip, &full_value)) {
```

Comment:
This if is always is true, otherwise we never come inside this function. Can we change it into an assert? I am fine to take care of it later also.

Reply:
I don't understand why this is always true. `__wt_row_leaf_value(page, rip, &full_value)` is only true for simple globally visible values.

Reply:
The `unpack->tw` is initialized with default values if the code flow doesn't enter into that if check. It is always entering into that if check and I feel there is no need to check for it.

Reply:
I'm not sure especially for overflow keys.

Reply:
I'm incline to just keep it as it is.

Reply:
You are right, this is always true as we need to fix the onpage value, which means the onpage value cannot be globally visible.

Reply:
I'll change that.